torch_np/_normalizations.py

```diff
@@ -96,7 +98,7 @@ def maybe_normalize(arg, parm, return_on_failure=_sentinel):
             raise exc from None
 
 
-def normalizer(_func=None, *, return_on_failure=_sentinel):
+def normalizer(_func=None, *, return_on_failure=_sentinel, mangle_result=True):
```

Comment:
```
>>> np.shape([[1], [2]])
(2, 1)
```

Reply:
Can you elaborate?

Reply:
np.shape is an example of a function which receives and array_like and returns a tuple of python scalars. Normalizing its return in a standard tuple-of-tensors way would give a tuple of arrays, `(array(2), array(1))`, instead of an expected `(2, 1)`. To prevent this, it's decorated with `normalizer(mangle_result=False)` [yes, _mangle_ should be renamed to _wrap_ ]

Reply:
My point is that we should respect the typing of the output and just map tensors to ndarrays. In this case the output has no tensors, so we return no ndarrays (it's the identity in this case).
If one given function returns a tuple of numbers and we'd want it to return a tuple of arrays (I don't think there's any example of this really, but still) we would wrap the tuple to a tuple of tensors in `_impl` and that's it.

Reply:
Ah, I see. So you want to have an extra type check, not just `isinstance(result, tuple)`, but also of the tuple elements. 
I did not think about it really, numpy's _scalar is an array_like, too_ is too heavily imprinted in my brain, apparently :-). Let me try it and see what breaks in the test suite.